src/loophole.coffee

```diff
@@ -2,19 +2,35 @@ vm = require 'vm'
 
 exports.allowUnsafeEval = (fn) ->
   previousEval = global.eval
+  callback = -> global.eval = previousEval
+  error = false
   try
     global.eval = (source) -> vm.runInThisContext(source)
-    fn()
+    fn(callback)
+
+  catch err
+    error = true
+    callback()
+
   finally
-    global.eval = previousEval
+    if !fn.length and !error
```

Comment:
If `fn.length is 0` I think you should throw an exception immediately and not run any code... there's no way they should be using this without a callback.